include/albatross/src/core/distribution.hpp

```diff
@@ -132,6 +132,11 @@ struct MarginalDistribution : public DistributionBase<MarginalDistribution> {
     return MarginalDistribution(scale * mean, scale * scale * covariance);
   }
 
+  MarginalDistribution operator[](std::size_t index) const {
```

Comment:
Nice! Should we move this to `DistributionBase<>` so it's available in `JointDistribution`s as well? (and maybe add a test to this suite: https://github.com/swift-nav/albatross/blob/master/tests/test_core_distribution.cc#L240)

Reply:
I can definitely make those changes.  It's a bit odd because for a `JointDistribution` the return type is still a `MarginalDistribution`.

Reply:
Yeah I see, I guess if you want `JointDistribution` to preserve it's type you could return `DistributionBase`. I'm fine either way though.